### PR TITLE
changing node exporter ignore list

### PR DESCRIPTION
**Pull request by prashbnair**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

The node exporter should not export data about veth interfaces. In case of the OVN, the regex was incorrect and was exporting veth metrics. This PR fixes that issue.

## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [x] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. Later this will be copied to the changelog file._

```release-note

```

### CODE REVIEW

File: jsonnet/kube-prometheus/components/node-exporter.libsonnet [hunk 1]
```diff
@@ -178,8 +178,8 @@ function(params) {
         // NOTE: ignore veth network interface associated with containers.
         // OVN renames veth.* to <rand-hex>@if<X> where X is /sys/class/net/<if>/ifindex
         // thus [a-z0-9] regex below
-        '--collector.netclass.ignored-devices=^(veth.*|[a-z0-9]+@if\\d+)$',
-        '--collector.netdev.device-exclude=^(veth.*|[a-z0-9]+@if\\d+)$',
+        '--collector.netclass.ignored-devices=^(veth.*|[a-f0-9]{15})$',
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/components/node-exporter.libsonnet:181]: Is there any other, more generic way of writing this? 

Relevant discussion in https://github.com/prometheus-operator/kube-prometheus/pull/1224#discussion_r661252578

Comment (prashbnair) [reply jsonnet/kube-prometheus/components/node-exporter.libsonnet:181]: The node exporter reads from devices from the /sys/class/net
https://github.com/prometheus/node_exporter#enabled-by-default
 and this only has 15 characters when using OVN.

Comment (paulfantom) [reply jsonnet/kube-prometheus/components/node-exporter.libsonnet:181]: Yes, correct. My question is about making this regex OVN-agnostic. I.e. can we make this applicable to other CNI implementations? 
If there is no generic way, let's go with what is in this PR.

Comment (prashbnair) [reply jsonnet/kube-prometheus/components/node-exporter.libsonnet:181]: My understanding is that veth.* will catch it all other CNI implementations. It is only in the case of OVN that it is being renamed.

**Pull request merged**